TypeCobol/AntlrGrammar/CobolCodeElements.g4

```diff
@@ -8261,6 +8262,13 @@ execStatementEnd: END_EXEC;
 
 //FOR SQL
 commitStatement: SQL_COMMIT;
+truncateStatement: SQL_TRUNCATE SQL_TABLE? truncateClause;
+truncateClause: (tableName=tableOrViewOrCorrelationName) (dropStorage|reuseStorage) (ignoreDeleteTriggers|restrictWhenDeleteTriggers) SQL_IMMEDIATE? ;
+dropStorage: SQL_DROP storage;
+reuseStorage: ({ string.Equals(CurrentToken.Text, "REUSE", System.StringComparison.OrdinalIgnoreCase) }? KeywordREUSE=UserDefinedWord) storage;
+storage: ({ string.Equals(CurrentToken.Text, "STORAGE", System.StringComparison.OrdinalIgnoreCase) }? KeywordSTORAGE=UserDefinedWord);
+restrictWhenDeleteTriggers: SQL_RESTRICT SQL_WHEN SQL_DELETE SQL_TRIGGER;
+ignoreDeleteTriggers: ({ string.Equals(CurrentToken.Text, "IGNORE", System.StringComparison.OrdinalIgnoreCase) }? KeywordIGNORE=UserDefinedWord) SQL_DELETE SQL_TRIGGER;
```

Comment:
The modelization here should be improved:
- no need for a `truncateClause` since it is mandatory to have a tableName
- first you need to define the context sensitive keywords: `STORAGE`, `REUSE`, `IGNORE` and `TRIGGERS`
- then I would go for:
  - `truncateStatement: SQL_TRUNCATE SQL_TABLE? (tableName=tableOrViewOrCorrelationName) storageManagementClause? deleteTriggersHandlingClause? SQL_IMMEDIATE?;`
  - `storageManagementClause: (SQL_DROP | reuse) storage;`
  - `deleteTriggersHandlingClause: (ignore | SQL_RESTRICT SQL_WHEN) SQL_DELETE triggers;`

The grammar must help us identify the semantics of the statement so here we define clauses based on the semantic of the different options. This is important because it will help us capture the options actually used by the developer when building the object. You must think about the building process when writing the grammar, for example if we want to capture a syntax property for the storage management of Truncate statement then it will be easier to look for a 'storageManagementClause' instead of checking for dropStorage or reuseStorage individually.